extensions/apps/extensions/src/components/pages/extension-release-manager/extension-release-manager-page.tsx

```diff
@@ -0,0 +1,328 @@
+
+  const showErrorNotification = React.useCallback((title: string) => {
+    uiEventsRef.current.next({
+      event: NotificationEvents.ShowNotification,
+      data: {
+        type: NotificationTypes.Error,
+        title,
+      },
+    });
+  }, []);
+
+  const draftExtensions: Extension[] = useMemo(() => {
+    try {
+      return JSON.parse(localStorage.getItem(`${DRAFT_EXTENSIONS}-${authenticatedDID}`)) || [];
+    } catch (error) {
+      showErrorNotification(error);
+    }
+  }, [authenticatedDID, showErrorNotification]);
+
+  const localExtensionData = draftExtensions.find(ext => ext.id === extensionId);
+
+  const {
+    data: appsByIdReq,
+    loading: loadingAppsByIdQuery,
+    error: appsByIdError,
+  } = useGetAppsByIdQuery({ variables: { id: extensionId } });
+
+  const publishedAppData = appsByIdReq?.node;
+
+  const extensionData = useMemo(() => {
+    if (publishedAppData && 'applicationType' in publishedAppData) {
+      return publishedAppData;
+    }
+    return localExtensionData;
+  }, [localExtensionData, publishedAppData]);
+
+  const draftReleases = useMemo(() => {
+    try {
+      return JSON.parse(localStorage.getItem(`${DRAFT_RELEASES}-${authenticatedDID}`)) || [];
+    } catch (error) {
+      showErrorNotification(error);
+    }
+  }, [authenticatedDID, showErrorNotification]);
+
+  const localRelease = draftReleases.find(release => release.applicationID === extensionId);
+
+  const {
+    data: appsReleasesReq,
+    loading: loadingAppsReleasesQuery,
+    error: appsReleasesError,
+    fetchMore,
+  } = useGetAppsReleasesQuery({
+    variables: {
+      first: 10,
+      filters: { where: { applicationID: { equalTo: extensionId } } },
+      sorting: { createdAt: SortOrder.Desc },
+    },
+  });
+
+  const appReleases = selectAppsReleases(appsReleasesReq);
+
+  const pageInfo = selectAppsReleasesPageInfo(appsReleasesReq);
+
+  const handleConnectButtonClick = () => {
+    navigateTo?.({
+      appName: '@akashaorg/app-auth-ewa',
+      getNavigationUrl: (routes: Record<string, string>) => {
+        return `${routes.Connect}?${new URLSearchParams({
+          redirectTo: `${baseRouteName}/release-manager/${extensionId}`,
+        }).toString()}`;
+      },
+    });
+  };
+
+  const handleTestReleaseNav = () => {
+    // if there is no local test release create it now
+    if (!localRelease) {
+      const newRelease = {
+        applicationID: extensionId,
+        version: '0.0.1',
+        description: 'Introduced the core functionality allowing developer to test.',
+        source: '',
+      };
+
+      localStorage.setItem(
+        `${DRAFT_RELEASES}-${authenticatedDID}`,
+        JSON.stringify([...draftReleases, newRelease]),
+      );
+    }
+    navigate({ to: '/release-manager/$extensionId/edit-test-release', params: { extensionId } });
+  };
+
+  const handlePublishReleaseNav = () => {
+    navigate({ to: '/release-manager/$extensionId/publish-release', params: { extensionId } });
```

Comment:
extensionId needs to be urlEncoded everywhere otherwise navigation involving and extension with a slash in it's name will not work - example: `mynamespace/my-extension`. You can use `encodeAppName` and `decodeAppName` available via `useRootComponentProps`

Note that when using the extensionId in a apollo hook you will need to use decodeAppName and when you do navigation use encodeAppName.

Reply:
ok, so the extensionId is the id from the model (not the extension name). then this should work.

Reply:
Ok, thanks, I will change this